core/drand_beacon_control.go

```diff
@@ -428,9 +427,16 @@ func (bp *BeaconProcess) runDKG(leader bool, group *key.Group, timeout uint32, r
 		"starting_beacon_time", finalGroup.GenesisTime, "now", bp.opts.clock.Now().Unix())
 
 	// beacon will start at the genesis time specified
-	go bp.StartBeacon(false)
+	// TODO (dlsniper): Old code launched StartBeacon in a goroutine.
```

Comment:
This is a potentially bigger change than it looks.
Please review/consider it before merging. I can undo it too and open an issue if desired.

Reply:
All the processes could provide close/error channels and the `DrandDaemon` just loops over a big select. 
I think more generally it would be nice to abstract a lot of the functionalities as discrete processes (beacon process, catchup process, dkg process, etc)

Reply:
Also right now creation of the `DrandDaemon` and the kick-off of its component parts (e.g. Http servers, beacon processing) is all intermixed in the `NewDrandDaemon` function - it'd be nice to separate them too

Reply:
Let's talk about it ;)

Reply:
So this wasn't changed in the end, correct?

Reply:
No changes as of this PR. I've opened a follow-up issue to document this: https://github.com/drand/drand/issues/1111